src/paulscode/android/mupen64plusae/CacheRomInfoFragment.java

```diff
@@ -204,4 +211,63 @@ public boolean IsInProgress()
     {
         return mInProgress;
     }
+    
+
+    public static File extractRomFile( File destDir, ZipEntry zipEntry, InputStream inStream )
+    {        
+        // Read the first 4 bytes of the entry
+        byte[] buffer = new byte[1024];
+        try
+        {
+            if( inStream.read( buffer, 0, 4 ) != 4 )
+                return null;
+        }
+        catch( IOException e )
+        {
+            Log.w( "GalleryActivity", e );
+            return null;
+        }
+        
```

Comment:
The fix was to remove some unnecessary checks in here.  The rest was just refactoring.